src/newwave/entity.mo

```diff
@@ -17,88 +17,275 @@ import Result "mo:base/Result";
+  */
+  public type EntityAttachedBridges = Types.Result<[Text], EntityAttachedBridgesErrors>;
+
+  /**
+   * Defines the errors for the public API when trying to retrieve an Entity ID
+  */
+  public type EntityIdErrors = {
+    #Unauthorized;
+    #EntityNotFound;
+    #Error;
+  };
+
+  /**
+   * Defines the result type for trying to retieve the ID of an entity from
+   * the public API
+  */
+  public type EntityIdResult = Types.Result<Text, EntityIdErrors>;
+
+  /**
+   * Defines the entity errors that can occur when trying to retrieve an entity
+  */
+  public type EntityErrors = {
+    #Unauthorized : Text;
+    #EntityNotFound;
+    #Error;
+  };
+
+  /**
+   * Defines the result type for trying to retrieve an Entity from
+   * the public API
+  */
+  public type EntityResult = Types.Result<Entity, EntityErrors>;
+
+  /**
+   * Stores entity specific settings
+  */
+  public class EntitySettings() {
+    var mainSetting : Text = "default";
+  };
+
+  /**
+   * The available entity types that can be used to describe an entity
+  */
+  public type EntityType = {
+    #BridgeEntity;
+    #Webasset;
+    #Person;
+    #Location;
+  };
+
+  /**
+   * Type that defines the attributes for an Entity
+  */
   public type Entity = {
-    internalId : Text;
+    /**
+     * The ID of the Entity that is used to store it in
+     * in the entity database
+    */
+    id : Text;
+    /**
+     * The timestamp in UTC (maybe) that the entity was created
+    */
     creationTimestamp : Nat64;
+    /**
+     * The original creator of the entity.
+    */
     creator : Principal;
+    /**
+     * The current owner of the entity
+    */
     owner : Principal;
-    settings : EntitySettings.EntitySettings;
-    entityType : EntityType.EntityType;
+    /**
+     * Settings for the entity
+    */
+    settings : EntitySettings;
+    /**
+     * The type that defines the entity
+    */
+    entityType : EntityType;
+    /**
+     * A human readable name? for the entity
+    */
     name : ?Text;
+    /**
+     * An owner defined description for what the entity is
+    */
     description : ?Text;
+    /**
+     * Keywords that are used to descripe the entity to
+     * enable more efficient lookup of the entity?
+    */
     keywords : ?[Text];
-    externalId : ?Text;
+    /**
+     * Unknown
+    */
     entitySpecificFields : ?Text;
+    /**
+     * Unknown
+    */
     listOfEntitySpecificFieldKeys : [Text];
-    // resolveRepresentedEntity : () -> T; // if possible, generic return value, otherwise probably Text
+
+    /**
+     * Contains all the bridge ids that originate from this
+     * Entity
+    */
+    fromIds : [Text];
+
+    /**
+     * Contains all the bridge ids that point to this entity
+    */
+    toIds : [Text];
   };
 
+  /**
+   * The initialization object is the fields provided by a user
+   * in order to create an entity. The rest of the fields are automatically
+   * created by Bebb
+  */
   public type EntityInitiationObject = {
```

Comment:
maybe make the id part of the initiation object as well (instead of passing it separately to the generate function below)?

Reply:
I see we're using the initiation object in the public interface and that's probably the reason you didn't add the id to the type